### PR TITLE
add event for when a direct download is triggered

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
Makes sense

File: apps/dav/lib/Events/DirectDownloadEvent.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace OCA\DAV\Events;
+
+use OCP\EventDispatcher\Event;
+use OCP\Files\File;
+
+/**
+ * @since 22.0.0
+ */
+class DirectDownloadEvent extends Event {
```
Comment (ChristophWurst) [inline apps/dav/lib/Events/DirectDownloadEvent.php:32]: BeforeDirectFileDownloadedEvent? https://docs.nextcloud.com/server/18/developer_manual/app/events.html#naming-scheme

Comment (icewind1991) [reply apps/dav/lib/Events/DirectDownloadEvent.php]: Went with `BeforeFileDirectDownloadedEvent`

**Review by ChristophWurst — APPROVED** [review]

File: apps/dav/lib/Direct/DirectFile.php [hunk 2]
```diff
@@ -43,9 +45,12 @@ class DirectFile implements IFile {
 	/** @var File */
 	private $file;
 
-	public function __construct(Direct $direct, IRootFolder $rootFolder) {
+	private $eventDispatcher;
+
+	public function __construct(Direct $direct, IRootFolder $rootFolder, IEventDispatcher $eventDispatcher) {
```
Comment (ChristophWurst) [inline apps/dav/lib/Direct/DirectFile.php:50]: > /home/runner/work/server/server/apps/dav/lib/Direct/DirectHome.php:86:11:error - TooFewArguments: Too few arguments for OCA\DAV\Direct\DirectFile::__construct - expecting 3 but saw 2

Comment (icewind1991) [reply apps/dav/lib/Direct/DirectFile.php:50]: fixed

**Pull request merged**